cvnp/cvnp.cpp

```diff
@@ -9,7 +9,59 @@ namespace cvnp
     namespace detail
     {
         namespace py = pybind11;
-        
+
+        // Translated from cv2_numpy.cpp in OpenCV source code
+        class CvnpAllocator : public cv::MatAllocator {
+        public:
+            CvnpAllocator() { stdAllocator = cv::Mat::getStdAllocator(); }
+            ~CvnpAllocator() {}
+
+            cv::UMatData* allocate(pybind11::array& a) const {
+                //int dims, const int* sizes, int type, size_t* step) const {
+                cv::UMatData* u = new cv::UMatData(this);
+                u->data = u->origdata = (uchar*)a.mutable_data(0);
+                // auto *strides = a.strides();
+                // for( int i = 0; i < dims - 1; i++ )
+                //     step[i] = (size_t)_strides[i];
+                // step[dims-1] = CV_ELEM_SIZE(type);
+                // u->size = sizes[0]*step[0];
+                u->size = a.size();
+                u->userdata = a.inc_ref().ptr();
+                return u;
```

Comment:
You should add 

```cpp
u->refcount = 1;
```

here. This would replace the strange call to `m.addref()` later